cloud_chatops/src/pr_reminder.py

```diff
@@ -0,0 +1,221 @@
+        self.CHANNEL = "temp-chatops"
+
+    def run_public(self, mention, channel=None) -> None:
+        """
+        This method calls the functions to post the reminder message and subsequent PR messages.
+        :param channel: Changes the channel to post the messages to.
+        :param mention: To mention the users in Slack or just post their name.
+        """
+        if channel:
+            self.CHANNEL = channel
+        reminder_message = self.post_reminder_message()
+        self.post_thread_messages(self.prs, reminder_message, mention)
+
+    def run_private(self, channel) -> None:
+        """
+        This method should send all open PRs to a specific user channel and only send their PRs.
+        :param channel: The chanel/user ID to send to and filter for.
+        """
+        self.CHANNEL = channel
+        reminder_message = self.post_reminder_message()
+        self.post_thread_messages(
+            self.prs, reminder_message, mention=False, private_user=channel
+        )
+
+    def post_reminder_message(self) -> WebClient.chat_postMessage:
+        """
+        This method posts the main reminder message to start the thread if PR notifications.
+        :return: The reminder message return object
+        """
+        reminder = self.client.chat_postMessage(
+            channel=self.CHANNEL,
+            text="Here are the outstanding PRs as of today:",
+        )
+        return reminder
+
+    def post_thread_messages(
+        self,
+        prs: Dict[str, List],
+        reminder_message: WebClient.chat_postMessage,
+        mention: bool,
+        private_user=None,
+    ) -> None:
+        """
+        This method collates each individual PR reminder message and calls a post method.
+        :param private_user: The user to filter messages for if chosen.
+        :param mention: To mention the users or not
+        :param prs: A list of PRs from GitHub
+        :param reminder_message: The reminder message object
+        """
+        private_count = 0
+        for repo in prs.values():
+            for pr in repo:
+                info = {
+                    "pr_title": f"{pr['title']} #{pr['number']}",
+                    "user": pr["user"]["login"],
+                    "url": pr["html_url"],
+                    "created_at": pr["created_at"],
+                    "channel": reminder_message.data["channel"],
+                    "thread_ts": reminder_message.data["ts"],
+                    "mention": mention,
+                    "draft": pr["draft"],
+                }
+                checked_info = self.check_pr(info)
+                if private_user:
+                    if info["user"] == private_user:
+                        private_count += 1
+                        self.send_thread(**checked_info)
+                    else:
+                        pass
+                else:
+                    self.send_thread(**checked_info)
+        if not private_count and private_user:
```

Comment:
I think it should be `and not private_user`, as the method should send no prs if there isn't a `private_user` **and** `private_count` is zero?

(I noticed the last post by Cloud ChatOps printed all the PRs and the send no PRs message in the same thread)

Reply:
I think the variable naming here is quite poor and easy to misunderstand. I will amend this and it should make sense. Private user is when the notifications are going only to a single person using the command. The private count is just if we found a PR belonging to the command executor